docs/developer_docs/developer_notes.md

```diff
@@ -0,0 +1,198 @@
+
+"Compilation" generates a `.ctf` (Component Technology File)
+which is a zip archive containing metainformation and encrypted m-files as described in detail below.
+
+There are some restrictions associated with the `ctf` archives:
+
+* A `ctf` archive generated by one Matlab release (e.g. R2020a)
+  cannot be read by the Matlab Compiler Runtime (MCR) for another release (e.g. R2022b)
+* You must compile all the m-files you need with a single run of `mcc`
+  e.g. you cannot run `mcc` on half the files, then `mcc` again on the other half
+  and manually combine the two sets of encrypted m-files in a single zipped `ctf`.
+
+`mcc` allows you to include any type of Matlab files including class definitions,
+but only specific named functions in the `mcc` command are exported and are callable by `feval`.
+The list of these "`public-functions`" is given in `.META/manifest.xml` and can be extracted
+by the Python function `libpymcr.Matlab.get_matlab_functions()`.
+
+This restriction does not apply to the compiled Matlab m-files which can internally call
+any Matlab system function or classes as well as all included m-files.
+We thus get around the "`public-functions`" restriction,
+and also allow things like class constructors to be called
+by defining a gateway function [`call.m`](../../src/call.m)
+which internally calls the Matlab `feval`.
+
+`call.m` in addition also:
+
+* Wraps and unwraps "old-style" Matlab classes
+  (defined only by `@`-files or directories not by `classdef`) which would otherwise be
+  interpreted as a `struct` and converted to a Python `dict` on output,
+  losing connection with their methods.
+  The wrapping is done by creating a `containers.Map` with one key-value pair where the
+  value is the old-style object and the key is `'wrapped_oldstyle_class'`.
+  A `containers.Map` is interpreted as a new-style `object` so is wrapped in a Python
+  opaque class (see type conversion below).
+* Unwraps Python callables and calls them using the `call_python` mex file.
+  The wrapping is done in `libpymcr::matlab_env` and involves creating a `struct` with two fields:
+  + `func_ptr` which holds the memory address (pointer) of the Python function.
+  + `converter` which holds the memory address (pointer) of the `libpymcr::pymat_converter` object
+  This is passed to the `call_python` mex file which converts them into a `PyObject`
+  and `pymat_converter` instance, executes the Python code and converts the results to Matlab format.
+
+Finally, there is a Python class, `libpymcr.MatlabProxyObject`
+which overloads various Python dunder methods with equivalent calls to Matlab methods
+to allow the "opaque" Matlab objects to be inspected from Python.
+
+
+## Type conversion
+
+The core of the module is the `libpymcr::pymat_converter` [class](../../src/type_converter.hpp)
+which converts data from Python to Matlab and vice versa, according to:
+
+| Python      |to| Matlab  ||  Matlab     |to| Python    |
```

Comment:
Don't think the table formatting is qworking as expected (at least using  `... > view file` in the top corner)
![image](https://user-images.githubusercontent.com/55979119/227583315-06528eb6-9a24-4af8-bf2c-1926a72d7b10.png)

Reply:
Yeah... I'm not sure what's happening :( I used a different online Markdown editor and it worked there...